vllm/model_executor/guided_decoding/outlines_decoding.py

```diff
@@ -65,8 +66,9 @@ async def get_outlines_guided_decoding_logits_processor(
         return None
 
     if global_thread_pool is None:
+        max_workers = os.cpu_count() or 2
```

Comment:
I would put an upper bound for this until we know it is necessary to use dozens of cores. Maybe minimum of core count and 8 or 16?

Reply:
sure, i'll cap it!

Reply:
I just capped it to 16, with a note that we may revisit if testing proves it helpful enough